ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_statistical.py

```diff
@@ -260,3 +260,42 @@ def test_corrcoef(
         y=x[1],
         rowvar=rowvar,
     )
+
+
+# bincount
+@handle_test(
+    fn_tree="functional.ivy.experimental.bincount",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=["int32"],
+        num_arrays=2,
+        shared_dtype=True,
+        min_num_dims=1,
+        max_num_dims=1,
+        min_dim_size=1,
+        max_dim_size=10,
+        min_value=0,
+        max_value=10,
+        allow_nan=False,
+    ),
+    test_gradients=st.just(False),
```

Comment:
the weights strategy should be generated within `@handle_test`, perhaps a composite would be easier to make here, here's a similar example that could give you some inspiration: https://github.com/unifyai/ivy/blob/38f579bd487fa555aa58b3ea1f2a5e918b88f41d/ivy_tests/test_ivy/test_functional/test_core/test_linalg.py#L1365
Essentially you want to generate your `x` array, and then generate the weights as either None, or int/float array of the same shape as `x` (use draw for this) and you should also generate an integer and pass it for `minlength`. Also there are differences with the dtype each backend outputs. I would recommend you modify each backend to cast the output to either the default ivy integer dtype, or to the dtype of `weights` if it's a float and not None